x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/types.ts

```diff
@@ -17,11 +17,13 @@ export const DEFAULT_META = {
   total: 0,
 };
 
-export const convertMetaToPagination = (meta: Meta) => ({
-  pageIndex: meta.from - 1,
-  pageSize: meta.size,
-  totalItemCount: meta.total,
-});
+export const convertMetaToPagination = (meta: Meta) => {
+  return {
+    pageIndex: meta.from / meta.size,
+    pageSize: meta.size,
+    totalItemCount: meta.total,
+  };
+};
 export const updateMetaPageIndex = (oldState: Meta, newPageIndex: number) => {
-  return { ...oldState, from: newPageIndex };
+  return { ...oldState, from: (newPageIndex - 1) * oldState.size };
```

Comment:
question: can newPageIndex ever be 0, and thus this return a negative number?

Reply:
i’ll double check that.

Reply:
refactored this, the change handler was adding 1 to the index to give base 1 pages instead of base 0.